src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -697,10 +735,14 @@ def __write_status_file(self):
             shutil.rmtree(self.complete_status_file_path)
 
         # Write complete status file <seq.no>.complete.status
-        self.env_layer.file_system.write_with_retry_using_temp_file(self.complete_status_file_path, '[{0}]'.format(json.dumps(complete_status_payload)), mode='w+')
+        status_file_payload_json_dumps = json.dumps(complete_status_payload)
+        self.env_layer.file_system.write_with_retry_using_temp_file(self.complete_status_file_path, '[{0}]'.format(status_file_payload_json_dumps), mode='w+')
+
+        if Constants.StatusTruncationConfig.TURN_ON_TRUNCATION:
+            status_file_payload_json_dumps = self.__check_file_size_and_timestamp_for_truncation(status_file_payload_json_dumps)
```

Comment:
Function should be renamed to reflect it's action. __check_file_size_and_timestamp_for_truncation() suggests it's only checking for file size and timestamp and NOT actually truncating

Reply:
do you have a better name for the method?

Reply:
Flagged for discussion if needed. @feng-j678 @rane-rajasi

Reply:
@feng-j678 IIUC, this function is not just delaying based on the timestamp but also **performing truncation** when applicable, which should also be a part of the function name.

Some suggestions for the function name here are: get_status_payload_post_truncation() or get_status_payload_with_truncated_patches()

This will be the superset/parent function, which can then have function calls within them to check for the filesize or timestamp or whether a delay is required. But those functions, should strictly be restricted to one action only, for eg: a function such as check_file_size() should only compute and return file size, timestamp_for_truncation() should only compute the timestamp and return it. is_truncation_delayed() returns whether to delay truncation.

Reply:
see new code changes, pending